libs/barista-components/chart/src/selection-area/selection-area-overlay-ref.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { OverlayRef } from '@angular/cdk/overlay';
+import { Observable, Subject } from 'rxjs';
+import { DtChartSelectionAreaContainer } from './selection-area-container';
+import { OnDestroy } from '@angular/core';
+
+export class DtChartSelectionAreaOverlayRef implements OnDestroy {
+  /** Subject that emits and completes when the overlay ref is destroyed */
+  destroy$ = new Subject<void>();
+
+  /** The ref of the selection area overlay */
+  overlayRef: OverlayRef;
+
+  /** Container reference of the selection area overlay */
+  containerRef: DtChartSelectionAreaContainer;
+
+  constructor(
+    overlayRef: OverlayRef,
+    containerRef: DtChartSelectionAreaContainer,
+  ) {
+    this.overlayRef = overlayRef;
+    this.containerRef = containerRef;
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
```

Comment:
You don't have Angular lifecycles in pure classes as it is not a part in the Angular Dependency Injection Tree.

You have to handle destroy on your own so please create a destroy method that is called in an `ngOnDestroy` of the wrapping Angular Component.

A class is only used to hold a data state in angular just a better object.